src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -752,11 +752,34 @@ def assign_college_simulants_guaridans(self, pop: pd.DataFrame) -> pd.DataFrame:
                     households_guardian_types[guardian_type]["race_ethnicity"] == race,
                     "reference_person_id",
                 ]
-                race_guardian_ids = self.randomness.choice(
-                    race_college_sims_with_guardian_idx,
-                    choices=race_reference_person_ids,
-                    additional_key=f"{race}_{guardian_type}_guardian_ids",
-                )
+                if (
+                    race_reference_person_ids.empty & len(race_college_sims_with_guardian_idx)
+                    > 0
+                ):
+                    # Get ids of for all other race reference persons if guardian type has no reference persons
```

Comment:
This comment is confusing; I was interpreting "for all other race reference persons" as somehow meaning you were trying to assign guardians of a different race ("all other race") which is exactly incorrect.